packages/blockchain/src/index.ts

```diff
@@ -939,30 +943,48 @@ export default class Blockchain implements BlockchainInterface {
         }
       }
 
-      // set total difficulty in the current context scope
-      if (this._headHeaderHash) {
-        currentTd.header = await this.getTotalDifficulty(this._headHeaderHash)
+      if (block._common.consensusType() === ConsensusType.ProofOfStake) {
+        if (block._common.hardfork() !== 'merge') {
+          throw new Error('cannot insert PoS block when merge hardfork not set')
+        }
+        // TODO: Check if POS block is descendant of valid block
```

Comment:
could you explain more what you mean by this?

Reply:
From the EIP

Add verification of block validity with respect to the PoS consensus rules, i.e. assert that there MUST be a corresponding event for the block to consider it valid. Such events are limited to the following:
A POS_CONSENSUS_VALIDATED event considering the block or **any of its ancestors as valid.**
A POS_CHAINHEAD_SET event, including the POS_BLOCK_FINALIZED form of this event, designating the block as the head of the canonical chain.

I'm not sure what exactly this would look like but some sort of check the block hash is either matched by that `POS_CONSENSUS_VALIDATED` event or else an ancestor block.

Reply:
I would rather assume that these things will implicitly happen in the client later on, e.g. once the client has received a `POS_CONSENSUS_VALIDATED` call by RPC or something it then calls the `put...` function on the blockchain class.

For now I guess it's ok if we skip this event related part completely on a first round, this will likely become more clear once we start testing with another (Eth 2.0) client.